core/src/SYCL/Kokkos_SYCL_Instance.cpp

```diff
@@ -75,7 +75,15 @@ void SYCLInternal::initialize(const sycl::device& d) {
       Kokkos::Impl::throw_runtime_exception(
           "There was an asynchronous SYCL error!\n");
   };
-  initialize(sycl::queue{d, exception_handler});
+  // FIXME SYCL+Cuda Apparently, submit_barrier doesn't quite work as expected
+  // for oneAPI 2023.0.0 on NVIDIA GPUs.
+#if defined(KOKKOS_IMPL_ARCH_NVIDIA_GPU)
+  initialize(
+      sycl::queue{d, exception_handler, sycl::property::queue::in_order()});
```

Comment:
I thought that was the default

Reply:
Is the "in order" property not the default?
If the answer is "no it depends on the device", then is that not something that we should then be specifying unconditionally?

Reply:
No, by default `sycl::queus` are out-of-order and I wanted to support that for flexibility in the implementation, e.g., when it comes to our `Graph` interface (which doesn't exploit out-of-order `queues`, yet).